strongbox-commons/src/main/java/org/carlspring/strongbox/io/ArtifactInputStream.java

```diff
@@ -1,78 +1,145 @@
 package org.carlspring.strongbox.io;
 
-import org.carlspring.commons.http.range.ByteRange;
-import org.carlspring.commons.io.MultipleDigestInputStream;
-import org.carlspring.commons.io.reloading.ReloadableInputStreamHandler;
-import org.carlspring.maven.commons.util.ArtifactUtils;
-import org.carlspring.strongbox.artifact.coordinates.ArtifactCoordinates;
-
+import java.io.FilterInputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.security.MessageDigest;
 import java.security.NoSuchAlgorithmException;
-import java.util.List;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.apache.commons.codec.digest.MessageDigestAlgorithms;
+import org.carlspring.commons.encryption.EncryptionAlgorithmsEnum;
+import org.carlspring.commons.util.MessageDigestUtils;
+import org.carlspring.strongbox.artifact.coordinates.ArtifactCoordinates;
 
 /**
  * @author mtodorov
  */
-public class ArtifactInputStream extends MultipleDigestInputStream
+public class ArtifactInputStream extends FilterInputStream
```

Comment:
This is a controversial situation which I told you earlier.
I meet controversial situation with AIS tooday, It is about ByteRangeInputStream.
I think that AIS should not extend ByteRangeInputStream, it must wrap it instead, if necessary.
It's of a series Aggregation vs Inheritance and Aggregation is better to use here.